### PR TITLE
Add PDF export

### PR DESCRIPTION
**Pull request by racell**
- Make new dialog for PDF export
- Used 'itext' library to create PDF file

This PR solve issue #304

### CODE REVIEW

File: app/src/main/res/layout/dialog_pdf_export.xml [hunk 1]
```diff
@@ -0,0 +1,150 @@
+<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (emartynov) [inline app/src/main/res/layout/dialog_pdf_export.xml]: Why do you need to duplicate similar layout for CSV layout? You need to change only title

File: app/src/main/java/org/glucosio/android/activity/MainActivity.java [hunk 2]
```diff
@@ -425,9 +428,11 @@ public void onClick(DialogInterface dialog, int which) {
                     } else {
                         dialog.dismiss();
                     }
-                } else {
+                } else if (which == 1) {
                     // Export to CSV
                     showExportCsvDialog();
+                } else {
+                    showExportPdfDialog();
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/activity/MainActivity.java]: Why do you need two different methods? They are almost identical

File: app/src/main/java/org/glucosio/android/activity/MainActivity.java [hunk 3]
```diff
@@ -528,6 +533,99 @@ public void onClick(View v) {
 
     }
 
+    public void showExportPdfDialog() {
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/activity/MainActivity.java]: Same comment about duplication

File: app/src/main/java/org/glucosio/android/activity/MainActivity.java [hunk 4]
```diff
@@ -528,6 +533,99 @@ public void onClick(View v) {
...
+                exportRangeButton.setChecked(!isChecked);
+                exportButton.setEnabled(true);
+            }
+        });
+
+        exportButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (validateExportDialog()) {
+                    exportPdfPresenter.onExportClicked(exportAllButton.isChecked());
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/activity/MainActivity.java]: This point is only one difference from another export dialog. Why not to generalise and reuse it?

File: app/src/main/java/org/glucosio/android/presenter/ExportPdfPresenter.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+                    mExportView.onExportStarted(readings.size());
+                    if (readings.isEmpty()) {
+                        isEmpty[0] = true;
+                        return null;
+                    }
+
+                    if (dirExists()) {
+                        Log.i("glucosio", "Dir exists");
+                        return ReadingToPdf.createPDFFile(mActivity, realm, readings, preferredUnit);
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/presenter/ExportPdfPresenter.java]: This line is only difference from original export presenter

File: app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java [hunk 6]
```diff
@@ -38,7 +38,7 @@
 
     public static String createCSVFile(Context context, Realm realm, final List<GlucoseReading> readings, String um) {
         try {
-            final File file = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) + "/glucosio", "glucosio_export_ " + System.currentTimeMillis() / 1000 + ".csv");
+            final File file = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) + "/glucosio", "glucosio_export_" + System.currentTimeMillis() / 1000 + ".csv");
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java]: Good catch!

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -159,7 +159,7 @@
     <string name="preferences_about">About</string>
     <string name="preferences_experimental_title" translatable="false">Here be dragons :/</string>
     <string name="preferences_experimental">These features are experimental and may not fully work and are for early testing by users. Remember that you can shake your phone to send us feedback.</string>
-    <string name="version" translatable="false">1.2.1 (Chutney)</string>
+    <string name="version" translatable="false">1.3.0 (Chutney)</string>
```
Comment (emartynov) [inline app/src/main/res/values/strings.xml]: Version bump should not be part of this functionality

File: wear/build.gradle [hunk 8]
```diff
@@ -29,7 +29,7 @@ android {
         applicationId "org.glucosio.android"
         minSdkVersion 21
         targetSdkVersion 25
-        versionCode 36
+        versionCode 38
```
Comment (emartynov) [inline wear/build.gradle]: Same comment about version code bumping

File: app/build.gradle [hunk 9]
```diff
@@ -34,12 +34,14 @@ android {
...
         targetSdkVersion 25
-        versionCode 36
+        versionCode 38
         versionName '1.3.0'
         applicationId 'org.glucosio.android'
 
         vectorDrawables.useSupportLibrary = true
         testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'
+
+        multiDexEnabled true
```
Comment (emartynov) [inline app/build.gradle]: This is not enough! You should also properly instantiate multidex in app

**Comment by racell** [comment]
@emartynov Thank You for your code review!

I thought this PDF export replace CSV export.
that's my misunderstanding.
so I leaved duplication.
duplication will be delete, I work in progress.

File: app/build.gradle [hunk 10]
```diff
@@ -34,12 +34,14 @@ android {
     defaultConfig {
         minSdkVersion 16
         targetSdkVersion 25
-        versionCode 36
+        versionCode 38
```
Comment (racell) [inline app/build.gradle]: I thought this change was caused by the android studio update.
if you think this version up is unnecessary, I will be revert this.

File: app/src/main/java/org/glucosio/android/activity/MainActivity.java (hunk 4 above)
Comment (racell) [inline app/src/main/java/org/glucosio/android/activity/MainActivity.java]: You have deleted this duplicate dialog

File: app/src/main/res/layout/dialog_pdf_export.xml (hunk 1 above)
Comment (racell) [inline app/src/main/res/layout/dialog_pdf_export.xml]: I have deleted this duplicate dialog.

File: app/src/main/res/values/strings.xml (hunk 7 above)
Comment (racell) [inline app/src/main/res/values/strings.xml]: Same about above comment

File: wear/build.gradle (hunk 8 above)
Comment (racell) [inline wear/build.gradle]: Same about above comment

**Comment by paolorotolo** [comment]
Hey @racell, any progress on this? I would like to see PDF export shipped in next release :)

**Comment by bkerensa** [comment]
@racell Do you have time to finish this?

**Comment by emartynov** [comment]
I'm closing since there is no activity. Sorry!

**Comment by bkerensa** [comment]
@racell any chance you want to work on this again?

**Comment by raacker** [comment]
@emartynov @bkerensa  I'm pretty sure PDF is not supported yet right? I'll ask him personally to continue on this

**Pull request closed without merging**